test/suites/dev/moonbase/test-precompile/test-precompile-pallet-xcm-2.ts

```diff
@@ -0,0 +1,416 @@
+import "@moonbeam-network/api-augment";
+import {
+  beforeAll,
+  describeSuite,
+  fetchCompiledContract,
+  expect,
+  customDevRpcRequest,
+} from "@moonwall/cli";
+import {
+  ALITH_ADDRESS,
+  BALTATHAR_ADDRESS,
+  alith,
+  baltathar,
+  createEthersTransaction,
+  PRECOMPILE_NATIVE_ERC20_ADDRESS,
+} from "@moonwall/util";
+import { encodeFunctionData, erc20Abi } from "viem";
+import {
+  expectEVMResult,
+  PARA_1000_SOURCE_LOCATION,
+  mockHrmpChannelExistanceTx,
+  ARBITRARY_ASSET_ID,
+  mockAssetBalance,
+  registerForeignAsset,
+  relayAssetMetadata,
+  RELAY_SOURCE_LOCATION_V4,
+} from "../../../../helpers";
+import type { AssetMetadata } from "../../../../helpers";
+import { ethers } from "ethers";
+
+const PRECOMPILE_PALLET_XCM_ADDRESS: `0x${string}` = "0x000000000000000000000000000000000000081A";
+
+export const para1000AssetMetadata: AssetMetadata = {
+  name: "PARA",
+  symbol: "PARA",
+  decimals: 12n,
+  isFrozen: false,
+};
+
+describeSuite({
+  id: "D012901",
+  title: "Precompiles - PalletXcm: Native fee",
+  foundationMethods: "dev",
+  testCases: ({ context, it }) => {
+    let foreignRelayAssetContract: ethers.Contract;
+    let foreignParaAssetContract: ethers.Contract;
+
+    const destinationPara = 1000;
+    const amountToSend = 100n;
+
+    beforeAll(async () => {
+      const balance = 200000000000000n;
+
+      // Register the asset
+      const { registeredAssetId: relayAssetId, contractAddress: relayAssetAddress } =
```

Comment:
I don't know if it suits the needs of this test, but there is a new helper that unifies `registerForeignAsset` and `mockAssetBalance` called `registerAndFundAsset`